File: src/main/cpp/length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        if (s.size() == 0) return 0;
        int begin = -1;
        for (int i = s.size() - 1; i >= 0; --i) {
            if (s.at(i) != ' ') {
                begin = i;
                break;
            }
        }
        if (begin == -1) return 0;
        int i = begin;
        for (; i >= 0; --i)  {
            if (s[i] == ' ') {
                break;
            }
        }
        return begin - i;
    }
};

File: src/main/java/LengthOfLastWord.java
public class LengthOfLastWord {
    public int lengthOfLastWord(String s) {
        if (s == null) return 0;
        boolean flag = false;
        int begin = s.length() -1 ;
        for (int i = s.length() - 1; i >= 0; --i) {
            if (s.charAt(i) != ' ') {
                flag = true;
                begin = i;
                break;
            }
        }
        if (flag == false) {
            return 0;
        }
        int i = begin;
        for (; i >= 0; i--) {
            if (s.charAt(i) == ' ') {
                break;
            }
        }
        return begin - i;
    }
}
